cpp/src/mlt/decode/int.cpp

```diff
@@ -0,0 +1,51 @@
+#include <mlt/decode/int.hpp>
+#include <mlt/decode/int_template.hpp>
+
+// from fastpfor/...
+#include <compositecodec.h>
+#include <fastpfor.h>
+#include <variablebyte.h>
+
+#include <cstdint>
+
+namespace mlt::decoder {
+
+struct IntegerDecoder::Impl {
+    FastPForLib::CompositeCodec<FastPForLib::FastPFor<8>, FastPForLib::VariableByte> codec;
+    // Reused storage for intermediate values
+    std::vector<std::uint32_t> buffer4;
+    std::vector<std::uint64_t> buffer8;
+};
+
+IntegerDecoder::IntegerDecoder()
+    : impl(std::make_unique<Impl>()) {}
+
+IntegerDecoder::~IntegerDecoder() noexcept = default;
+
+template <typename T>
+std::vector<T>& IntegerDecoder::getTempBuffer() noexcept {
+    // Can we make these covariant and eliminate the casts?
+    constexpr auto is4 = std::is_trivially_assignable_v<typename decltype(impl->buffer4)::reference, T>;
+    constexpr auto is8 = std::is_trivially_assignable_v<typename decltype(impl->buffer8)::reference, T>;
+    static_assert(is4 || is8);
+    if constexpr (is4) {
+        return reinterpret_cast<std::vector<T>&>(impl->buffer4);
```

Comment:
Can you cast a `std::vector<std::uint32_t>` to a `std::vector<std::int32_t>`?

Reply:
I should revisit that anyway...

Reply:
Intermediate results all use the contents of a `uint8_t` vector now.